src/main/java/cubicchunks/world/column/Column.java

```diff
@@ -710,7 +714,8 @@ public boolean isTerrainPopulated() {
 		//with cubic chunks the whole column is never fully generated,
 		//So some heuristic is needed to tell vanilla is generator is populated here
 		//for now - tell it that it is if any cube is populated
-		return this.cubeMap.all().stream().anyMatch(c -> c.getGeneratorStage().isLastStage());
+		
+		return this.cubeMap.all().stream().anyMatch(c -> (c.getGeneratorStage() == GeneratorStage.LIVE));
```

Comment:
As I said, stage itself should know if it's final stage or not. It can be a final method that internally just compares to LIVE, but I don't think making the LIVE stage public is a good idea considering that all other stages aren't public.